infra.bs

```diff
@@ -282,8 +283,106 @@ a simple wrapper around another one.)
 <dfn ignore>load a classic script</dfn> given <var>url</var>, return the result of performing the
 internal script-loading algorithm given <var>url</var> and "<code>classic</code>".
 
-<p class="XXX">We would also like to give guidance on named and optional parameters; see
-<a href="https://github.com/whatwg/infra/issues/320">issue #320</a> for ongoing discussion.
+
+<h3 id=algorithm-params>Parameters</h3>
+
+<p>Algorithm parameters are usually listed sequentially, in the fashion described in
+[[#algorithm-declaration]]. However, there are some more complicated cases.</p>
+
+<p>Algorithm parameters can be optional, in which case the algorithm declaration must list them as
+such, and list them after any non-optional parameters. They can either be given a default value, or
+the algorithm body can check whether or not the argument was given. Concretely, use the following
+forms:
+
+<p class=exemplary-prose>&hellip; an optional [type] <var ignore>[parameter]</var> &hellip;
+
+<p class=exemplary-prose>&hellip; an optional [type] <var ignore>[parameter]</var> (default [default
+value]) &hellip;
+
+<div class=example id=example-algorithm-optional-positional-params>
+ <p class=allow-2119>To <dfn ignore id=example-navigate-algo-positional>navigate</dfn> given a
+ resource <var ignore>resource</var>, an optional string <var ignore>navigationType</var>, and an
+ optional boolean <var ignore>exceptionsEnabled</var> (default false):
+
+ <ol class=brief>
+  <li>&hellip;
+  <li>If <var ignore>navigationType</var> was given, then do something with
+  <var ignore>navigationType</var>.
+  <li>&hellip;
+ </ol>
+</div>
+
+<p>To call algorithms with such optional positional parameters, the optional argument values can be
+omitted, but only the trailing ones.
+
+<div class=example id=example-algorithm-optional-positional-params-calling>
+ <p>Call sites to the previous example's algorithm would look like one of:
+
+ <ul class=brief>
+  <li><a href=#example-navigate-algo-positional>Navigate</a> to <var ignore>resource</var>.
+
+  <li><a href=#example-navigate-algo-positional>Navigate</a> to <var ignore>resource</var> with
+  "<code>form submission</code>".
+
+  <li><a href=#example-navigate-algo-positional>Navigate</a> to <var ignore>resource</var> with
+  "<code>form submission</code>" and true.
```

Comment:
Feels like we should call out positional booleans as bad practice.

Reply:
If we really want to do that I guess that would further argue for special casing booleans in some ways.

Reply:
Yeah, I dunno, it's OK-ish especially since it's all hypertext, but I agree generally it's not going to be a good time, and you probably want your booleans to be named.

Reply:
It seems based on the discussion we should change this too, right? To only have examples with recommended practice.

Reply:
Maybe. I'm unsure because it's part of the "narrative flow" of the section: it shows the algorithm evolving from unclear and inflexible optional positional arguments, to clear and flexible optional named arguments.

We could instead specifically explain that not only is this non-ideal because of the ordering restrictions, but also because of the presence of non-named boolean parameters?

Reply:
Yeah that sounds good to me. I suspect that would work for @jyasskin too.

Reply:
Done.

Reply:
Yep, thanks. Note that this example-of-things-we-don't-want-people-to-do winds up looking the same as all the other examples-of-things-to-do. Vanilla bikeshed supports a `heading="Bad Example"` attribute to make it a little clearer, but that attribute doesn't work with the [WHATWG style sheet](https://resources.whatwg.org/spec.css)'s `.example::before { content: 'Example'; }` rule.